pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -33,6 +33,13 @@ type Constraints struct {
 	// pod tolerations on a per-node basis.
 	// +optional
 	Taints Taints `json:"taints,omitempty"`
+	// StartupTaints are taints that may be applied to nodes upon startup either by the provisioner or a daemonset,
+	// but which are expected to be removed automatically within a short period of time. These are commonly used by
+	// daemonsets to allow initialization.  A startup taint on the provisioner does not cause that taint to be added to
+	// a node. If it appears on a node by also being configured via Taints or by being added by an external mechanism
+	// it will be ignored for provisioning purposes.
+	// +optional
+	StartupTaints Taints `json:"startupTaints,omitempty"`
```

Comment:
I agree this was confusing.  I've left it named as 'StartupTaints' for now, but I've modified it so that it now actually causes the taints to be applied to the node.    You'll just list the taint as a startup taint, we put it on the node but ignore it for scheduling and assume someone else will remove it.